move `lint` as dev dependency

MrSquaare opened the pull request:
Fixes #79

There are a lot of lint errors reported by `flutter analyze` (not introduced by this PR, this is already the case on `master`)
Should we also run ~~`flutter format` and perhaps~~ `dart fix`? I have run `flutter format` to fix Dart formatter reports.

This PR should fix Pass static analytics and Support up-to-date dependencies as reported by https://pub.dev/packages/styled_widget/score

MrSquaare committed:
build: move lint as dev dependency

MrSquaare committed:
build: bump lint to v2

MrSquaare committed:
style: run flutter format

ReinBentdal commented:
Hi, thanks for your contribution. I haven't been an active flutter developer for some time so limited resources spent on this package.
Please also do dart fix if you think it will improve. Will happily accept PR

MrSquaare committed:
refactor: run dart fix

- apply manually fixes for cast_nullable_to_non_nullable and prefer_final_locals

MrSquaare commented:
> Hi, thanks for your contribution. I haven't been an active flutter developer for some time so limited resources spent on this package. Please also do dart fix if you think it will improve. Will happily accept PR

Hi!
I ran `dart fix` and applied some fix manually (no auto-fix). There are still some lint reports (nothing very important, others related to TODO, so I prefer not to touch them)
Tell me if it's good for you.

I see that there have been some feature (#77) not yet released; will the next release be deployed when this PR is merged?

ReinBentdal commented:
I see i was a bit quick in accepting the PR. Could you please fix the following issues:
- in animated_text and animated_icon there should be no member fields (change for exampele this.text to String text in _StyledAnimatedTextContainer)
- Styled.dart -> styled.dart

I see the linter no longer likes classes with only static members. Do you have any suggestions for a solution?
